### PR TITLE
Initial support for multi-label classification.

### PR DESCRIPTION
**Pull request by trivialfis**
* Add support for sklearn prediction function.
* Add basic documents.
* Add tests.

Related: https://github.com/dmlc/xgboost/issues/2087 .

### CODE REVIEW

File: doc/tutorials/multioutput.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+################
+Multiple Outputs
+################
+
+.. versionadded:: 1.6
+
+Starting from version 1.6, XGBoost has experimental support for multi-output regression
```
Comment (trivialfis) [inline doc/tutorials/multioutput.rst:7]: Mention that only Python package is supported at the moment.

File: doc/tutorials/multioutput.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+and multi-label classification.  For the terminologies please refer to the `scikit-learn
+user guide <https://scikit-learn.org/stable/modules/multiclass.html>`_.
```
Comment (trivialfis) [inline doc/tutorials/multioutput.rst:9]: Provide a brief explanation

File: python-package/xgboost/sklearn.py [hunk 3]
```diff
@@ -1374,9 +1382,13 @@ def predict(
             # If output_margin is active, simply return the scores
             return class_probs
 
-        if len(class_probs.shape) > 1:
+        if len(class_probs.shape) > 1 and self.n_classes_ != 2:
             # turns softprob into softmax
             column_indexes: np.ndarray = np.argmax(class_probs, axis=1)  # type: ignore
+        elif len(class_probs.shape) > 1 and class_probs.shape[1] != 1:
```
Comment (trivialfis) [inline python-package/xgboost/sklearn.py:1388]: Is there a way to simplify the logic?

**Review by RAMitchell — APPROVED** [review]
